services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Exclude.java

```diff
@@ -0,0 +1,210 @@
+   * <p>
+   * {@link DirectionsCriteria#PROFILE_WALKING}: No excludes supported
+   * <p>
+   * {@link DirectionsCriteria#PROFILE_CYCLING}: {@link DirectionsCriteria#EXCLUDE_FERRY}
+   *
+   * @return a list of strings matching one
+   *   of the {@link DirectionsCriteria.ExcludeCriteria} exclusions
+   */
+  @Nullable
+  public abstract List<String> criteria();
+
+  /**
+   * Exclude certain points from routing. The default is to not exclude anything.
+   *
+   * @return a list of points excluded from the routing.
+   */
+  @Nullable
+  public abstract List<Point> points();
+
+  /**
+   * Use this builder to build an {@link Exclude} object.
+   */
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    /**
+     * Exclude certain road types from routing. The default is to not exclude anything from the
+     * profile selected. The following exclude flags are available for each profile:
+     * <p>
+     * {@link DirectionsCriteria#PROFILE_DRIVING}: One of {@link DirectionsCriteria#EXCLUDE_TOLL},
+     * {@link DirectionsCriteria#EXCLUDE_MOTORWAY}, or {@link DirectionsCriteria#EXCLUDE_FERRY}.
+     * <p>
+     * {@link DirectionsCriteria#PROFILE_DRIVING_TRAFFIC}: One of
+     * {@link DirectionsCriteria#EXCLUDE_TOLL}, {@link DirectionsCriteria#EXCLUDE_MOTORWAY}, or
+     * {@link DirectionsCriteria#EXCLUDE_FERRY}.
+     * <p>
+     * {@link DirectionsCriteria#PROFILE_WALKING}: No excludes supported
+     * <p>
+     * {@link DirectionsCriteria#PROFILE_CYCLING}: {@link DirectionsCriteria#EXCLUDE_FERRY}
+     *
+     * @param criteria a list of exclude criteria
+     *   matching {@link DirectionsCriteria.ExcludeCriteria}
+     * @return this builder.
+     */
+    public abstract Builder criteria(@Nullable List<String> criteria);
+
+    /**
+     * Exclude certain points from routing. The default is to not exclude anything.
+     *
+     * @param points a list of points to exclude.
+     * @return this builder.
+     */
+    public abstract Builder points(@Nullable List<Point> points);
+
+    /**
+     * Builds the object.
+     *
+     * @return a new instance of {@link Exclude}
+     */
+    public abstract Exclude build();
+  }
+
+  @Nullable
+  static Exclude fromUrlQueryParameter(@Nullable String queryParameter) {
+    if (queryParameter == null || Objects.equals(queryParameter, "")) {
+      return null;
+    }
+
+    ArrayList<Point> points = new ArrayList<>();
+    ArrayList<String> criteria = new ArrayList<>();
+
+    List<String> excludeItems = ParseUtils.parseToStrings(queryParameter, ",");
+    if (excludeItems != null) {
+      for (String excludeItem : excludeItems) {
+        if (excludeItem.startsWith("point(") && excludeItem.endsWith(")")) {
+          Point point = parsePoint(excludeItem);
+          points.add(point);
+        } else if (excludeItem.contains("(") && excludeItem.contains(")")) {
+          // Ignoring the unexpected type of data. Update the library to get support of new types
```

Comment:
Does this mean that you can't provide "exclude points" using the old `@Deprecated` APIs?

Reply:
You can. This line for different case:
```kotlin
  @Test
  public void parseUnexpectedTypeOfParameterMixedWithPointAndCriteria() {
    Exclude exclude = Exclude.fromUrlQueryParameter("ferry,new(test),point(0.5 -0.5)");
    assertNotNull(exclude);
    assertEquals(1, exclude.criteria().size());
    assertEquals(1, exclude.points().size());
  }
```
Directions API team may add new data formats for exclude. I think the best way would be to just ignore them during the parsing.

Reply:
resolving discussion as the code we're discussing has already been removed